File: app/src/main/java/cat/xojan/random1/injection/module/HomeModule.java
package cat.xojan.random1.injection.module;

import android.app.Activity;

import cat.xojan.random1.domain.interactor.ProgramDataInteractor;
import cat.xojan.random1.presenter.HomePresenter;
import cat.xojan.random1.viewmodel.PodcastsViewModel;
import cat.xojan.random1.viewmodel.ProgramsViewModel;
import cat.xojan.random1.viewmodel.SectionsViewModel;
import dagger.Module;
import dagger.Provides;

@Module
public class HomeModule {

    private final Activity mActivity;

    public HomeModule(Activity activity) {
        mActivity = activity;
    }

    @Provides
    HomePresenter provideHomePresenter(ProgramDataInteractor programDataInteractor) {
        return new HomePresenter(mActivity, programDataInteractor);
    }

    @Provides
    ProgramsViewModel provideProgramsViewModel(ProgramDataInteractor programDataInteractor) {
        return new ProgramsViewModel(programDataInteractor);
    }

    @Provides
    PodcastsViewModel providePOdcastsViewModel(ProgramDataInteractor programDataInteractor) {
        return new PodcastsViewModel(programDataInteractor);
    }

    @Provides
    SectionsViewModel provideSectionsViewModel(ProgramDataInteractor programDataInteractor) {
        return new SectionsViewModel(programDataInteractor);
    }
}

File: app/src/main/java/cat/xojan/random1/injection/module/HomeModule.kt
package cat.xojan.random1.injection.module

import android.app.Activity
import cat.xojan.random1.domain.interactor.PodcastDataInteractor
import cat.xojan.random1.domain.interactor.ProgramDataInteractor
import cat.xojan.random1.feature.browser.BrowserViewModel
import cat.xojan.random1.feature.home.HomeViewModel
import dagger.Module
import dagger.Provides

@Module
class HomeModule(private val activity: Activity) {

    @Provides
    fun provideBrowserViewModel(podcastInteractor: PodcastDataInteractor,
                                programInteractor: ProgramDataInteractor): BrowserViewModel {
        return BrowserViewModel(podcastInteractor, programInteractor)
    }

    @Provides
    fun provideHomeViewModel(podcastInteractor: PodcastDataInteractor): HomeViewModel {
        return HomeViewModel(podcastInteractor)
    }
}